object/types.proto

```diff
@@ -1,134 +1,128 @@
 syntax = "proto3";
+
 package object;
+
 option go_package = "github.com/nspcc-dev/neofs-api-go/object";
 option csharp_namespace = "NeoFS.API.Object";
 
 import "refs/types.proto";
 import "service/verify.proto";
 import "storagegroup/types.proto";
-import "github.com/gogo/protobuf/gogoproto/gogo.proto";
 
-option (gogoproto.stable_marshaler_all) = true;
+// Header groups the information about the NeoFS object.
+message Header {
+    // Main groups mandatory information about the object.
+    // Message fields are presented in all NeoFS objects.
+    message Main {
+        // PayloadLength carries length of the object payload.
+        // Each object has a fixed payload length since it's immutable.
+        uint64 PayloadLength = 1;
+        // Address carries object address in the NeoFS system.
+        // It encapsulates the object and the container identifiers.
+        refs.Address Address = 2;
+        // OwnerID carries identifier the object owner in a binary format.
+        bytes OwnerID = 3;
+    }
 
-message Range {
-    // Offset of the data range
-    uint64 Offset = 1;
-    // Length of the data range
-    uint64 Length = 2;
-}
+    // Main carries the main part of the header.
+    // Main MUST NOT be NULL.
+    Main main = 1;
```

Comment:
Does letter case matter here?

Reply:
Yes, `Main` field name declaration causes compilation error. However, such a message field is generated by the compiler in the public field of the structure. I researched this issue a bit and found this approach valid according to the [docs](https://developers.google.com/protocol-buffers/docs/proto3#nested).